instrumentation/net/http/otelhttp/client.go

```diff
@@ -0,0 +1,58 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelhttp
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+var otelDefaultClient = &http.Client{Transport: NewTransport(http.DefaultTransport)}
```

Comment:
Maybe we should expose this client as a default client for these wrappers, just like Go `http` lib.

```go
// DefaultClient is the default Client and is used by Get, Head, and Post.
var DefaultClient = &Client{}
```

Therefore, end-users can use `otelhttp.DefaultClient.Do()` as well.

Reply:
I went the other direction, actually.  Because the wrapped transport captures (e.g.) propagators, this could lead to unwanted ordering issues, such as 
```
Http.Get("/usesOldPropagator")
global.SetPropagators(newProps)
Http.Get("/shouldUseNewButOnlyIfWeDidNotCacheTheOld")
```
I discovered this by writing the unit tests and being surprised by the behavior.  We could offer `otelhttp.NewDefaultClient()` as a substitute?

Reply:
I'm wary of the construction of a new client for every request.  That seems like it could lead to performance issues as the client's connection pool cannot be re-used and the overhead of constructing a new client and transport must be incurred at every request rather than amortized over multiple requests.

Reply:
I agree with what @Aneurysm9 said about the performance. Adding a comment to the `DefaultClient` to notify users would be enough?

e.g., Some changes in `otel` might not affect the `DefaultClient`, such as setting a new global propagator.

Reply:
OK, I reverted back everything to an (exported) `DefaultClient`.  Thanks for the comments!